provisioning/transport/http/src/http_transport.ts

```diff
@@ -31,12 +31,16 @@ export class Http  extends EventEmitter implements ProvisioningTransportHandlers
     this._pollingHandlers.setTransportOptions(options);
   }
 
-  registerX509(registrationId: string, authorization: X509, forceRegistration: boolean, callback: (err?: Error, assignedHub?: string, deviceId?: string, responseBody?: any, result?: any) => void): void {
-    this._stateMachine.register(registrationId, authorization, {'registrationId' : registrationId}, forceRegistration, (err, responseBody, result) => {
+  registerX509(registrationId: string, auth: X509, forceRegistration: boolean, callback: (err?: Error, registrationResult?: X509RegistrationResult, body?: any, result?: any) => void): void {
+     this._stateMachine.register(registrationId, auth, {'registrationId' : registrationId}, forceRegistration, (err, responseBody, result) => {
       if (err) {
-        callback(err, null, null, responseBody, result );
+        callback(err, null, responseBody, result );
       } else {
-        callback(err, responseBody.registrationStatus.assignedHub, responseBody.registrationStatus.deviceId, responseBody, result);
+        let registrationResult: X509RegistrationResult = {
+          deviceId: responseBody.registrationStatus.deviceId,
+          assignedHub: responseBody.registrationStatus.assignedHub
+        };
```

Comment:
we shouldn't have to cast this - our interface should be there only to document what the service is sending back and we should be able to pass the registrationResult object to the callback directly.